src/main/scala/battleships/Grid.scala

```diff
@@ -0,0 +1,57 @@
+package battleships
+
+import battleships.Ship.Ship
+
+class Grid(s: Int) {
+  val grid = Array.ofDim[Ship](s, s)
+  val size = s
+
+  var lastx: Int = -1
+  var lasty: Int = -1
+
+  def printGrid(): Unit = {
+    var s: Ship = null
+    System.out.println()
+    for (i <- 0 to size) {
+      if (i == 0) {
+        print("  ")
+        for (k <- 0 to size) {
+          print(s"$k ")
+        }
+        println()
+      }
+
+      for (j <- 0 to size) {
+        s = grid(i)(j)
+        if (j == 0) print(s"$i ")
+        if (i == lasty && j == lastx) print("X ")
+        if (s == null)                print("~ ")
+        if (s == Ship.PATROLBOAT)     print("P ")
+        if (s == Ship.BATTLESHIP)     print("B ")
+        if (s == Ship.SUBMARINE)      print("S ")
+        if (s == Ship.DESTROYER)      print("D ")
+        if (s == Ship.CARRIER)        print("C ")
+      }
+      println()
+    }
+    println()
+  }
+
+  def placeShip(s: Ship, x: Int, y: Int, direction: String): Boolean = {
+    if (grid(y)(x) == null) {
+      if (direction.toLowerCase == "left")  for (i <- 0 to s.getSize) grid(y)(x - i) = s
```

Comment:
You're doing the same action 4 times "direction.toLowerCase" and "i <- 0 to s.getSize" could have them as fields so that you wouldn't need to do the same computation numerous times